ext/sockets/tests/gh17921.phpt

```diff
@@ -0,0 +1,20 @@
+--TEST--
+GH-16267 - overflow on socket_strerror argument
+--EXTENSIONS--
+sockets
+--INI--
+memory_limit=-1
```

Comment:
Why did you add this INI setting?

Reply:
... I have Allowed memory size warning otherwise...

Reply:
Hm but these functions all return false :thinking: i.e. they shouldn't reach the code that allocates the memory?
Actually, you adjusted the checks incorrectly though after my review, so you will actually reach this allocation code

Reply:
So ... isn't ZSTR_MAX_LEN == ZEND_ULONG_MAX ? length is a zend_long

Reply:
Seems I was wrong adding the ini setting only.

Reply:
> So ... isn't ZSTR_MAX_LEN fit for zend_ulong ? length is a zend_long. ZSTR_MAX_LEN > ZEND_LONG_MAX

Right, my bad, then this is fine